frame/support/src/traits/tokens/fungibles.rs

```diff
@@ -265,25 +265,52 @@ pub trait Create<AccountId>: Inspect<AccountId> {
 
 /// Trait for providing the ability to destroy existing fungible assets.
 pub trait Destroy<AccountId>: Inspect<AccountId> {
-	/// The witness data needed to destroy an asset.
-	type DestroyWitness;
-
-	/// Provide the appropriate witness data needed to destroy an asset.
-	fn get_destroy_witness(id: &Self::AssetId) -> Option<Self::DestroyWitness>;
-
-	/// Destroy an existing fungible asset.
-	/// * `id`: The `AssetId` to be destroyed.
-	/// * `witness`: Any witness data that needs to be provided to complete the operation
-	///   successfully.
+	/// Start the destruction an existing fungible asset.
+	/// * `id`: The `AssetId` to be destroyed. successfully.
 	/// * `maybe_check_owner`: An optional account id that can be used to authorize the destroy
-	///   command. If not provided, we will not do any authorization checks before destroying the
+	///   command. If not provided, no authorization checks will be performed before destroying
 	///   asset.
+	fn start_destroy(id: Self::AssetId, maybe_check_owner: Option<AccountId>) -> DispatchResult;
+
+	/// Destroy all accounts associated with a given asset.
+	/// `destroy_accounts` should only be called after `start_destroy` has been called, and the
+	/// asset is in a `Destroying` state
 	///
-	/// If successful, this function will return the actual witness data from the destroyed asset.
-	/// This may be different than the witness data provided, and can be used to refund weight.
-	fn destroy(
-		id: Self::AssetId,
-		witness: Self::DestroyWitness,
-		maybe_check_owner: Option<AccountId>,
-	) -> Result<Self::DestroyWitness, DispatchError>;
+	/// - `id`: The identifier of the asset to be destroyed. This must identify an existing asset.
```

Comment:
minor note, the formatting is different, with what you have above, I think the rust docs can be used as a reference https://doc.rust-lang.org/rust-by-example/meta/doc.html

I also might be important because polkadot js displays the docs in UI, but not sure what it exactly expects